File: 136-single-number/single-number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        // calculate num frequency
        unordered_map<int, int> mp;
        for (int i=0; i<nums.size(); i++) {
            if (mp.find(nums[i]) == mp.end()) {
                mp[nums[i]] = 1;
            }
            else {
                mp[nums[i]]++;
            }
        }
        // search for 1
        for (auto x: mp) {
            if (x.second == 1)
                return x.first;
        }
        return -1;
    }
};

File: 136-single-number/single-number.java
import java.util.*;

class Solution {
    public int singleNumber(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int sumSet = 0;
        int sumNums = 0;

        // get unique value
        for (int num : nums) {
            if (!set.contains(num)) {
                set.add(num);
                sumSet += num;
            }
            // acculate sum
            sumNums += num;
        }

        // 2* (a1 + a2 + ... + ak + ak+1) - 2*(a1 + a2 +... + ak) + ak-1
        return 2 * sumSet - sumNums;
    }
}